DonoGear/app/src/main/java/com/example/donogear/actionpages/HomePageFragment.java

```diff
@@ -27,6 +64,135 @@ public HomePageFragment() {
+        announcementAdapter.notifyDataSetChanged();
+        System.out.println("Initialize end");
+
+        learnMore = view.findViewById(R.id.learn_more_id);
+        learnMore.setOnClickListener(new View.OnClickListener() {
+            public void onClick(View v) {
+                Intent browserIntent = new Intent(
+                        Intent.ACTION_VIEW,
+                        Uri.parse(getString(R.string.donowebsite)));
+                startActivity(browserIntent);
+            }
+        });
+
+        myInterestsText = view.findViewById(R.id.myinterests_text);
+        myInterestsRecyclerView = view.findViewById(R.id.card_view_myinterests_recycler_list);
+        LinearLayoutManager horizontalInterestsLayoutManager = new LinearLayoutManager(getContext(), LinearLayoutManager.HORIZONTAL, false);
+        myInterestsRecyclerView.setLayoutManager(horizontalInterestsLayoutManager);
+        myInterestsItemList = new ArrayList<>();
+        myInterestsItemAdapter = new MyInterestsItemAdapter(activity, activity.listOfItems);
+        myInterestsItemList = activity.listOfItems;
+        if (myInterestsItemList.size() > 0) {
+            myInterestsText.setVisibility(View.VISIBLE);
+        }
+        else {
+            myInterestsText.setVisibility(View.GONE);
+        }
+        System.out.println(myInterestsItemList.size());
+        myInterestsItemAdapter.setItemList(myInterestsItemList);
+        System.out.println(myInterestsItemAdapter.getItemCount());
+        myInterestsRecyclerView.setAdapter(myInterestsItemAdapter);
+        myInterestsItemAdapter.setClickListener(this, MY_INTERESTS);
+        myInterestsItemAdapter.notifyDataSetChanged();
+    }
+
+
+    /**
+     * Filter item list and set trending item list
+     * @param itemDetailsList - list of all items
+     * @return - list of items which are trending
+     */
+    private List<ItemDetails> displayTrendingItems(List<ItemDetails> itemDetailsList) {
+        for (ItemDetails itemDetails: itemDetailsList) {
+            if (itemDetails.isTrending && itemDetails.listOfImages.size() > 0) {
+                System.out.println("Item added");
+                trendingItemList.add(itemDetails);
+            }
+        }
+        return trendingItemList;
+    }
+
+    /**
+     * Overriding the interface method to perform action for onClick on any trending item. Will call
+     * the next trending item details activity
+     * @param view - clicked view
+     * @param position - position of the item clicked
+     */
+    @Override
+    public void onItemClick(View view, int position, String type) {
+        ItemDetails item;
+        if (type.equals(TRENDING)) {
+            item = trendingItemList.get(position);
+        }
+        else {
+            item = myInterestsItemList.get(position);
+        }
+//        System.out.println(item.itemName);
+        Intent intent = new Intent(activity, ProductDetails.class);
+        intent.putExtra("item_details", item);
```

Comment:
Should probably make constants or private variables for "item_details" and "typeOfSearch"